geopandas/tests/test_geom_methods.py

```diff
@@ -486,6 +500,22 @@ def test_buffer_args(self):
             expected = original.buffer(10, **args)
             assert calculated.equals(expected)
 
+    def test_buffer_distance_array(self):
+        original = GeoSeries([self.p0, self.p0])
+        expected = GeoSeries(
+            [Polygon(((6, 5), (5, 4), (4, 5), (5, 6), (6, 5))),
+             Polygon(((10, 5), (5, 0), (0, 5), (5, 10), (10, 5))),
+             ])
+        calculated = original.buffer(np.array([1, 5]), resolution=1)
+        for _expected, _calculated in zip(expected, calculated):
+            assert _calculated.almost_equals(_expected)
```

Comment:
You can use `assert_geoseries_equal` here instead of iterating through the result/expected.